### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by 1alessandro1** [hieplpvip/AppleBacklightSmoother#2]
Hi, thanks for your work, the idea is pretty neat. 
Since it seems like the merge with whatevergreen won't happen, could you provide few lines of documentation (if someone who already has SSDT-PNLF in ACPI needs something else to change to use this kext/ or he/she just has to build it and copy it to the /EFI/OC/Kexts folder)

Thanks in advance

**Comment by 1alessandro1** [comment]
![image](https://user-images.githubusercontent.com/46293832/95018792-2f983d00-0662-11eb-80f7-f26563a28586.png)
Hi, I'm trying to build this kext but no luck so far. I've cloned and build Lilu.kext (debug) and copied it in the master directory alongside MacKernelSDK but the build still fails. Could you provide some releases just to test this please?

**Comment by hieplpvip** [comment]
I have pushed the fix for compiling with new Lilu. Please try again.

About documentation, I plan to do it in a few days. Basically, you only need to inject AppleBacklightSmoother and replace WhateverGreen's SSDT-PNLF with this one's. If you have Coffee Lake, you also need to turn off WhateverGreen's backlight patch with boot argument `igfxcflbklt=0`

**Comment by 1alessandro1** [comment]
Thanks a lot. I made a workaround before by commenting those lines, but I'll clone the repo and build again
I'll report back

**Comment by 1alessandro1** [comment]
> If you have Coffee Lake

I'm on Whiskey Lake (i5 8265U) does this apply as well as CoffeeLake?

**Comment by 1alessandro1** [comment]
My device is

    | |   | |   "AAPL,ig-platform-id" = <0900a53e>

**Comment by 1alessandro1** [comment]
This is part of my DSDT, should I remove it or merge your SSDT-PNLF?

![image](https://user-images.githubusercontent.com/46293832/95019529-3759e080-0666-11eb-9990-e8166f7d4409.png)

**Comment by hieplpvip** [comment]
You need to understand why SSDT-PNLF exists. Its purpose is to make AppleBacklight loaded. Only these lines are needed to do that:
```
Device(_SB.PCI0.GFX0.PNLF)
{
    Name(_ADR, Zero)
    Name(_HID, EisaId("APP0002"))
    Name(_CID, "backlight")
    // _UID is set depending on device ID to match profiles in WhateverGreen
    // 14: Arrandale/Sandy/Ivy
    // 15: Haswell/Broadwell
    // 16: Skylake/KabyLake
    // 17: custom LMAX=0x7a1
    // 18: custom LMAX=0x1499
    // 19: CoffeeLake 0xffff
    Name(_UID, 0)
    Name(_STA, 0x0B)
}
```

However, only so is not enough to make backlight working. There is a mismatch in PWM MAX between macOS and our laptop, so we need to fix that, either by patching macOS (which AppleBacklightSmoother and WhateverGreen's CoffeeLake patch does) or set PWM MAX in hardware to match that of macOS (which the traditional [SSDT-PNLF](https://github.com/acidanthera/WhateverGreen/blob/master/Manual/SSDT-PNLF.dsl) does). In addition, we also need to set UID to match profiles defined in [WhateverGreen](https://github.com/acidanthera/WhateverGreen/blob/f31896efc9eb28e0eaad694e9268c7ae5e6f9134/WhateverGreen/kern_weg.cpp#L32-L89)). But each platform needs a different value. If you look into SSDT-PNLF (both from this repo and WhateverGreen), you would see it contains device-id of various IGPU. They are used for automatically detecting platform and injecting corresponding value.

SSDT-PNLF from WhateverGreen also set PWM MAX, but as I said before, it's not needed for AppleBacklightSmoother as it patches macOS to work with all PWM MAX.

**TLDR:** If you want to keep things simple, no need to use SSDT-PNLF from this repo. You only need to change _UID to match your platform. You have Whiskey Lake, so you should set it to 19 (or 0x13).

**Comment by 1alessandro1** [comment]
Ok thanks for clarifying.

I'll edit _UID shown in the picture: (now set to 0x0A) to 0x13

![image](https://user-images.githubusercontent.com/46293832/95020080-3e362280-0669-11eb-8e73-1a1adb37cd37.png)


p.s. of course this device is under _SB.PCI0, do I need to move it to PCI0.GFX0?

![image](https://user-images.githubusercontent.com/46293832/95020096-50b05c00-0669-11eb-90f4-582bcd737e23.png)

**Comment by 1alessandro1** [comment]
I left the PNLF device where it was, also, is there any precaution on when this kext shoud be loaded? Such as right after WhateverGreen or is that ok to be after AppleALC or in general I can put this at the end?

**Comment by hieplpvip** [comment]
Injection order doesn't matter as long as it is injected after Lilu

**Comment by 1alessandro1** [comment]
Ok, about the location of the device, is it fine under Device(_SB.PCI0.PNLF) or does it require to be under Device(_SB.PCI0.GFX0.PNLF) ?

**Comment by hieplpvip** [comment]
I'm not sure about that

**Comment by 1alessandro1** [comment]
@hieplpvip

Sorry for asking again, but some time ago you said:
 
> you also need to turn off WhateverGreen's backlight patch with boot argument igfxcflbklt=0

Does this apply to Whiskey Lake as well (since its a derivative of CoffeeLake) or it is not necessary for the i5 8265U?
